src/network/gossip.rs

```diff
@@ -391,12 +462,45 @@ impl SnapchainGossip {
         }
     }
 
+    pub fn handle_contact_info(&mut self, contact_info: ContactInfo) {
+        // TODO(aditi): We might want to persist peers and reconnect to them on restart
+        let contact_info_body = contact_info.body.unwrap();
+        if self
+            .swarm
+            .connected_peers()
+            .find(|peer_id| PeerId::from_bytes(&contact_info_body.peer_id).unwrap() == **peer_id)
+            .is_some()
+        {
+            return;
+        }
+
+        if contact_info_body.network() != self.fc_network {
+            return;
+        }
+
+        if contact_info_body.snapchain_version != FARCASTER_VERSION {
+            return;
+        }
```

Comment:
We should also return early for validators?

Reply:
I changed the code so that we only call this function for read nodes. There's no need for validators to autodiscover peers.